pkg/services/preferences/preferences.go

```diff
@@ -0,0 +1,76 @@
+package preferences
+
+import (
+	"context"
+	"strings"
+
+	"github.com/grafana/grafana/pkg/models"
+	pstore "github.com/grafana/grafana/pkg/services/preferences/store"
+	"github.com/grafana/grafana/pkg/services/sqlstore"
+	"github.com/grafana/grafana/pkg/setting"
+)
+
+type Service interface {
+	GetPreferencesWithDefaults(ctx context.Context, query *models.GetPreferencesWithDefaultsQuery) (*models.Preferences, error)
+	GetPreferences(ctx context.Context, query *models.GetPreferencesQuery) (*models.Preferences, error)
+	SavePreferences(ctx context.Context, query *models.SavePreferencesCommand) error
+}
+
+type ServiceImpl struct {
+	PreferenceStore pstore.Store
+}
+
+func ProvideService(cfg *setting.Cfg, sqlstore sqlstore.Store) Service {
+	return &ServiceImpl{PreferenceStore: &pstore.StoreImpl{
```

Comment:
Shouldn't `PreferencesStore` have a constructor?

Reply:
yep, could be added